### PR TITLE
Add GS method to question a single client

### PR DESCRIPTION
**Pull request by ldpl**
Adds a GSGoal.QuestionClient method that works exactly like GSGoal.Question except that it only asks a single client.

As usual, simple GS for testing: [testclientquestion.tar.gz](https://github.com/OpenTTD/OpenTTD/files/1947211/testclientquestion.tar.gz)

### CODE REVIEW

File: src/goal.cpp [hunk 1]
```diff
@@ -243,17 +247,35 @@ CommandCost CmdGoalQuestion(TileIndex tile, DoCommandFlag flags, uint32 p1, uint
 {
 	uint16 uniqueid = (GoalType)GB(p1, 0, 16);
 	CompanyID company = (CompanyID)GB(p1, 16, 8);
-	byte type = GB(p1, 24, 8);
+#ifdef ENABLE_NETWORK
+	ClientIndex client = (ClientIndex)GB(p1, 16, 8);
+#endif
+	byte type = GB(p1, 24, 2);
```
Comment (frosch123) [inline src/goal.cpp:253]: Comment says "type" is in bits 24-26 (3 bits), code says 24-25 (2 bits).

File: src/goal.cpp [hunk 2]
```diff
@@ -243,17 +247,35 @@ CommandCost CmdGoalQuestion(TileIndex tile, DoCommandFlag flags, uint32 p1, uint
...
+	bool is_client = (p1 >> 31);
```
Comment (frosch123) [inline src/goal.cpp]: Please use HasBit().
We avoid implcit int->bool conversions.

File: src/goal.cpp [hunk 3]
```diff
@@ -243,17 +247,35 @@ CommandCost CmdGoalQuestion(TileIndex tile, DoCommandFlag flags, uint32 p1, uint
...
 
 	if (_current_company != OWNER_DEITY) return CMD_ERROR;
 	if (StrEmpty(text)) return CMD_ERROR;
-	if (company != INVALID_COMPANY && !Company::IsValidID(company)) return CMD_ERROR;
+	if (is_client) {
+#ifdef ENABLE_NETWORK
+		if (!NetworkClientInfo::IsValidID(client)) return CMD_ERROR;
+#endif
```
Comment (frosch123) [inline src/goal.cpp]: #else
return CMD_ERROR;

**Review by frosch123 — DISMISSED** [review]

**Review by TrueBrain — APPROVED** [review]

**Comment by TrueBrain** [comment]
Thank you, and keep them coming!

**Pull request merged**